tracer/src/Datadog.Trace/Debugger/Symbols/SymbolExtractor.cs

```diff
@@ -0,0 +1,718 @@
+
+                classMethods[index] = methodScope;
+                index++;
+            }
+        }
+
+        private SourcePdbInfo GetMethodSourcePdbInfo(Model.Scope methodScope)
+        {
+            var startLine = int.MaxValue;
+            var endLine = 0;
+            int? startColumn = null;
+            int? endColumn = null;
+            string? sourceFile = null;
+            for (int i = 0; i < methodScope.Scopes.Count; i++)
+            {
+                var scope = methodScope.Scopes[i];
+                if (startLine > scope.StartLine && scope.StartLine > 0)
+                {
+                    startLine = scope.StartLine;
+                }
+
+                if (endLine < scope.EndLine)
+                {
+                    endLine = scope.EndLine;
+                }
+
+                sourceFile ??= scope.SourceFile;
+
+                if (!scope.LanguageSpecifics.HasValue)
+                {
+                    continue;
+                }
+
+                if (startColumn == null || startColumn > scope.LanguageSpecifics.Value.StartColumn)
+                {
+                    startColumn = scope.LanguageSpecifics.Value.StartColumn;
+                }
+
+                if (endColumn == null || endColumn < scope.LanguageSpecifics.Value.EndColumn)
+                {
+                    endColumn = scope.LanguageSpecifics.Value.EndColumn;
+                }
+            }
+
+            if (startLine == int.MaxValue)
+            {
+                startLine = UnknownMethodStartLine;
+            }
+
+            if (endLine == 0)
+            {
+                endLine = UnknownMethodEndLine;
+            }
+
+            return new SourcePdbInfo { StartLine = startLine, EndLine = endLine, Path = sourceFile, StartColumn = startColumn, EndColumn = endColumn };
+        }
+
+        protected virtual Model.Scope CreateMethodScope(TypeDefinition type, MethodDefinition method)
+        {
+            // arguments
+            var argsSymbol = GetArgsSymbol(method);
+
+            // closures
+            var closureScopes = GetClosureScopes(type, method);
+            var methodAttributes = method.Attributes & StaticFinalVirtual;
+            var methodLanguageSpecifics = new LanguageSpecifics
+            {
+                ReturnType = method.DecodeSignature(new TypeProvider(false), 0).ReturnType,
+                AccessModifiers = (method.Attributes & MethodAttributes.MemberAccessMask) > 0 ? new[] { _methodAccess[Convert.ToUInt16(method.Attributes & MethodAttributes.MemberAccessMask)] } : null,
+                Annotations = methodAttributes > 0 ? new[] { _methodAttributes[Convert.ToUInt16(methodAttributes)] } : null
+            };
+
+            var methodScope = new Model.Scope
+            {
+                ScopeType = SymbolType.Method,
+                Name = MetadataReader.GetString(method.Name),
+                LanguageSpecifics = methodLanguageSpecifics,
+                Symbols = argsSymbol,
```

Comment:
`argsSymbol` might have empty entries (see line 594) - will it cause a problem down the road?

Reply:
Should be ok to put null whenever we need to. We will skip these in the uploaded json.

Reply:
Okay by me. Resolving.